### PR TITLE
Add a reference to can-log/dev to mustache-core; write a dev log test

### LINKED ISSUE #328: ReferenceError: dev is not defined
`dev` is not defined before it’s used here: https://github.com/canjs/can-stache/blob/086c8835c0da7983ffd0a004464604d26a497fd2/src/mustache_core.js#L184

### PR DESCRIPTION
**Pull request by bmomberger-bitovi**
Fixes #328

### CODE REVIEW

File: src/mustache_core.js [hunk 1]
```diff
@@ -12,6 +12,7 @@ var frag = require("can-util/dom/frag/frag");
 var attr = require("can-util/dom/attr/attr");
 var canSymbol = require("can-symbol");
 var canReflect = require("can-reflect");
+var dev = require("can-log/dev/dev");
```
Comment (phillipskevin) [inline src/mustache_core.js]: how was this working before?

File: test/stache-test.js [hunk 2]
```diff
@@ -6290,6 +6290,22 @@ function makeTest(name, doc, mutation) {
 		});
 	});
 
+	test("partials warn on missing context (#328)", function() {
```
Comment (phillipskevin) [inline test/stache-test.js]: this should be a dev only test.

Comment (bmomberger-bitovi) [reply test/stache-test.js]: Ugh, I should have caught that.

File: src/mustache_core.js (hunk 1 above)
Comment (bmomberger-bitovi) [reply src/mustache_core.js]: No test hit the block of code where it failed.

**Pull request merged**